search/search_repository.go

```diff
@@ -535,12 +567,29 @@ func parseFilterString(ctx context.Context, rawSearchString string) (criteria.Ex
 			"err":             err,
 			"rawSearchString": rawSearchString,
 		}, "failed to unmarshal raw search string")
-		return nil, errors.NewBadParameterError("expression", rawSearchString+": "+err.Error())
+		return nil, nil, errors.NewBadParameterError("expression", rawSearchString+": "+err.Error())
 	}
 	q := Query{}
 	parseMap(fm, &q)
 
-	return q.generateExpression()
+	fm2 := map[string]interface{}{}
+	// Parsing/Unmarshalling JSON encoding/json
+	err = json.Unmarshal([]byte(rawSearchString), &fm2)
```

Comment:
Why do we have to unmarshal the `rawSearchString` a second time around? It was already done and the errors where also already caught just a few lines above. Can't we just re-use that?

Reply:
Done. 04aa0e30c312e0f21ce1894185999e69f51beaa7